internal/ext/wasm/wasm.go

```diff
@@ -245,72 +195,56 @@ func (r *Runner) Invoke(ctx context.Context, method string, args any, reply any,
 		return fmt.Errorf("failed to encode codegen request: %w", err)
 	}
 
-	engine := wasmtime.NewEngine()
-	module, err := r.loadModule(ctx, engine)
+	cacheDir, err := cache.PluginsDir()
 	if err != nil {
-		return fmt.Errorf("loadModule: %w", err)
+		return err
 	}
 
-	linker := wasmtime.NewLinker(engine)
-	if err := linker.DefineWasi(); err != nil {
+	cache, err := wazero.NewCompilationCacheWithDir(filepath.Join(cacheDir, "wazero"))
+	if err != nil {
 		return err
 	}
 
-	dir, err := os.MkdirTemp(os.Getenv("SQLCTMPDIR"), "out")
+	wasmBytes, err := r.loadBytes(ctx)
 	if err != nil {
-		return fmt.Errorf("temp dir: %w", err)
+		return fmt.Errorf("loadModule: %w", err)
 	}
 
-	defer os.RemoveAll(dir)
-	stdinPath := filepath.Join(dir, "stdin")
-	stderrPath := filepath.Join(dir, "stderr")
-	stdoutPath := filepath.Join(dir, "stdout")
-
-	if err := os.WriteFile(stdinPath, stdinBlob, 0755); err != nil {
-		return fmt.Errorf("write file: %w", err)
-	}
+	config := wazero.NewRuntimeConfig().WithCompilationCache(cache)
+	rt := wazero.NewRuntimeWithConfig(ctx, config)
+	defer rt.Close(ctx)
 
-	// Configure WASI imports to write stdout into a file.
-	wasiConfig := wasmtime.NewWasiConfig()
-	wasiConfig.SetArgv([]string{"plugin.wasm", method})
-	wasiConfig.SetStdinFile(stdinPath)
-	wasiConfig.SetStdoutFile(stdoutPath)
-	wasiConfig.SetStderrFile(stderrPath)
+	// TODO: Handle error
```

Comment:
There is `Instantiate` if you'd like to return the error. Though I think any failure here would be a programming bug, not non-determinstic